File: ruby/examples/post_invoices_pay.rb
require_relative '../request.rb'

# Example of POST /invoices/pay

# This request pays all invoices given by invoice ids

PAYMENT_METHOD_ID = ENV['PAYMENT_METHOD_ID']
INVOICE_ID        = ENV['INVOICE_ID']

params = {
  'payment_token': PAYMENT_METHOD_ID,
  'invoice_ids': [INVOICE_ID]
}.to_json

BaseRequestRoute.request(:post, '/invoices/pay', body: params)

File: csharp/examples/PostInvoicesPay.cs
using System;
namespace Examples.examples
{
    // Example of POST /invoices/pay
    
    public class PostInvoicesPay : BaseRequest
    {
        private string _paymentToken;
        private string[] _invoiceIds;

        public PostInvoicesPay()
		{
            _paymentToken = Environment.GetEnvironmentVariable("PAYMENT_METHOD_ID");
            _invoiceIds = Environment.GetEnvironmentVariable("INVOICE_ID")?.Split(',');
            Console.WriteLine($"PostInvoicesPay: _paymentToken={_paymentToken} _invoiceIds={_invoiceIds}");
        }

        public override string SendRequest()
        {
            var body = Newtonsoft.Json.JsonConvert.SerializeObject(new
            {
                payment_token = _paymentToken,
                invoice_ids = _invoiceIds,
            });

            return PostRequest("invoices/pay", body);
        }
    }
}
